### PR TITLE
Mark record_component annotations as referenced

### PR DESCRIPTION
**Pull request by theresa-m**
Signed-off-by: Theresa Mammarella <Theresa.T.Mammarella@ibm.com>

### CODE REVIEW

**Comment by theresa-m** [comment]
For regular fields support for the `Contended` annotation is handled partly in `walkAnnotations`. I'm not sure if that applies here since record component fields are supposed to be final.

**Comment by pshipton** [comment]
jenkins test sanity,extended zlinux jdk11

File: runtime/bcutil/ClassFileOracle.cpp [hunk 1]
```diff
@@ -568,11 +568,19 @@ ClassFileOracle::walkRecordComponents(J9CfrAttributeRecord *attrib)
...
+					walkAnnotations(recordComponentAnnotations->numberOfAnnotations, recordComponentAnnotations->annotations, 0);
+				}
+				_recordComponentsInfo[i].annotationsAttribute = recordComponentAnnotations;
 				break;
 			}
 			case CFR_ATTRIBUTE_RuntimeVisibleTypeAnnotations: {
-				_recordComponentsInfo[i].typeAnnotationsAttribute = (J9CfrAttributeRuntimeVisibleTypeAnnotations *) recordComponentAttr;
+				J9CfrAttributeRuntimeVisibleTypeAnnotations *recordComponentTypeAnnotations = (J9CfrAttributeRuntimeVisibleTypeAnnotations *)recordComponentAttr;
+				if (0 == recordComponentTypeAnnotations->rawDataLength) {
+					walkTypeAnnotations(recordComponentTypeAnnotations->numberOfAnnotations, recordComponentTypeAnnotations->typeAnnotations);
```
Comment (DanHeidinga) [inline runtime/bcutil/ClassFileOracle.cpp]: The knownAnnotations argument is for tracking specialAnnotations.  It shouldn't be set unless the code needs to do something with the "found" annotations - and this code isn't as it isn't checking the return value
```suggestion
					walkTypeAnnotations(recordComponentTypeAnnotations->numberOfAnnotations, 0);
```

**Comment by DanHeidinga** [comment]
> For regular fields support for the `Contended` annotation is handled partly in `walkAnnotations`. I'm not sure if that applies here since record component fields are supposed to be final. Thoughts?

I wouldn't worry about it here.  Contended is mostly for use in JUC and is only respected for bootclasspath classes.  With modularity in 9, the annotation has moved to an internal package and is harder to access.  

All reasons to suspect it won't be used in Records

File: runtime/bcutil/ClassFileOracle.cpp (hunk 1 above)
Comment (theresa-m) [reply runtime/bcutil/ClassFileOracle.cpp]: It should be correct. Only `walkAnnotations` has the `knownAnnotations` parameter.

Comment (DanHeidinga) [reply runtime/bcutil/ClassFileOracle.cpp]: Sorry, my bad.  I misread the code

**Comment by DanHeidinga** [comment]
@theresa-m can you port this to the 0.20.0 branch?

**Comment by theresa-m** [comment]
Port to 0.20: https://github.com/eclipse/openj9/pull/8856

**Pull request merged**